src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -459,29 +456,30 @@ void BulletRigidObject::activateCollisionIsland() {
   // first query overlapping pairs of the current object from the most recent
   // broadphase to collect relevant simulation islands.
 
-  std::set<int> overlappingSimulationIslands = {thisColObj->getIslandTag()};
+  // each index represents an island tag present - default is -1, so add one.
+  std::bitset<65536> overlappingSimIslands;
```

Comment:
Magic Number! Can we derive it somehow? Or at the very least put some detailed comments on why this number was chosen (if arbitrary).

Reply:
Yeah, I hate it too.  It's a magic # in bullet - not even a constant there that I could harvest. :(

Reply:
Arbitrary :) [64k should be enough for everybody!](https://quoteinvestigator.com/2011/09/08/640k-enough/)

I'd write `64*1024` personally instead and add a note that this is just 8kB of memory and should be sufficient even for huge scenes (assuming that's true).

Reply:
[Here it is in bullet btCollisionObject.cpp constructor](https://github.com/bulletphysics/bullet3/blob/2a0f811c3bb3c65810eb2c095724b20b2f8b8dd4/src/BulletCollision/CollisionDispatch/btCollisionObject.cpp#L31)

Reply:
Oh, you mean the size of the bitset.  I'll add a comment saying that we expect to never have more than 65536 collision islands.